e2e/src/account/app.page.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * @license
+ * Copyright (c) 2018 THREEANGLE SOFTWARE SOLUTIONS SRL
+ * Available under MIT license webFrame/LICENSE
+ */
+
+import { IAccountCredentials, PAGE_URL } from 'app-shared/core';
+import { browser, by, Key, WebElement } from 'protractor';
+
+export class AppPage {
```

Comment:
rename to AppPageTestability.
For these types of classes follow this convention. It makes it obvious what it is, and for any component you have it will be easy to spot is there is any testability layer implemented.

Reply:
This is the default naming pattern for Angular e2e tests and it's using Martin Fowler's `PageOpbject`. The Protractor and Angular guidelines currently recommend the PageObject in order to organize the tests. And this is the naming pattern:
* a `*Page` suffix for the class
* a `.page.ts` ending for the file name

More info:
* https://www.protractortest.org/#/page-objects
* https://material.angular.io/guide/using-component-harnesses - which references:
  * https://martinfowler.com/bliki/PageObject.html

And actually, the `AppPage` should be renamed to `AccountPage` if we stick to that terminology for now. The only problem is that it's a bit misleading when you need to test multiple pages in a single test suite and you put the tests for all the different pages into a single "Page Object". This happens if you want to test the login and logout feature in a single `.e2e-spec.ts` file.

---

An alternative would be to implement the new Test Harnesses available in the Angular Component Development Kit, which have support for unit tests and for e2e tests. In Angular Material v9 all tests have been rewritten to use this new API:

> https://material.angular.io/guide/using-component-harnesses  
> **Which kinds of tests can use harnesses?**  
> The Angular CDK's component harnesses are designed to work in multiple different test environments. Support currently includes Angular's Testbed environment in Karma unit tests and Protractor end-to-end (e2e) tests. You can also support additional environments by creating custom extensions of the CDK's HarnessEnvironment and TestElement classes.

But in the end, we'll have to impelment more tests to find the suitable pattern for our project.

Reply:
We discussed this offline. For reusable testing components/pages, we will suffix it with Testability, it's more easy to search. Object is very generic, and you can have classes/components that use that term and are not related to testing.
Examples:
'DataTable' -> 'DataTableTestability'
'Autocomplete' -> 'AutocompleteTestability'.